pkgs/tools/package-management/nix/common.nix

```diff
@@ -207,6 +210,21 @@ stdenv.mkDerivation {
   passthru = {
     inherit boehmgc;
 
+    tests = lib.optionalAttrs enableStatic {
```

Comment:
you could alternatively use `allowedReferences = []` https://nixos.org/manual/nix/stable/expressions/advanced-attributes.html

Reply:
I tried that but it's actually tricky. I didn't find a way to only apply that rule to the "out" output, and not the "dev" one.

Reply:
> I tried that but it's actually tricky. I didn't find a way to only apply that rule to the "out" output, and not the "dev" one.

that's now possible with __structuredAttrs support in nixpkgs

for how to do it see https://nixos.org/manual/nix/stable/release-notes/rl-2.2.html